plumbing/api.go

```diff
@@ -204,7 +196,7 @@ func (api *API) MessagePreview(ctx context.Context, from, to address.Address, me
 // MessageQuery calls an actor's method using the most recent chain state. It is read-only,
 // it does not change any state. It is use to interrogate actor state. The from address
 // is optional; if not provided, an address will be chosen from the node's wallet.
-func (api *API) MessageQuery(ctx context.Context, optFrom, to address.Address, method string, params ...interface{}) ([][]byte, *exec.FunctionSignature, error) {
+func (api *API) MessageQuery(ctx context.Context, optFrom, to address.Address, method string, params ...interface{}) ([][]byte, error) {
```

Comment:
Ok, this is a pretty big change, so I want to explain it a bit. Apologies for the wall of text.

At the time of this commit, we're only using the method signature returned from `MessageQuery` in three places:

- `getLastUsedSectorID` method in `node/node.go`
- `isBootstrapMinerActor` method in `protocol/storage/miner.go`
- `getActorSectorCommitments` method in `protocol/storage/miner.go`

We also use the `MessageQuery` method and discard the method signature in over 36 different places. (I'm not going to list them all.) The method signature of a message query seems to be largely irrelevant on most invocations, since the calling code typically knows what to expect back in terms of an encoded response.

There may be some argument that by using a method signature and decoding with it we can avoid brittleness in the call sites when changing the method signatures themselves, but I'm inclined to believe that said brittleness can never be completely avoided because, at some point, we will almost always have to work with the typed object returned in some meaningful way, so if we change function signatures, we'll end up breaking these method invocations anyway. Using the method signature like this at best will only ever hide these problems, not solve them.

Additionally, because `MessageQuery` returns the method signature, `plumbing/msg` has a hard dependency on `plumbing/mthdsig` which contradicts our expressed goal of not creating dependencies between plumbing packages. Further, the *only* places the `mthdsig` package is used are here and in the `ActorGetSignature` plumbing method, which essentially does the exact same thing as `mthdsig`. This makes me think `mthdsig` should just be part of `actor` plumbing instead and `MessageQuery` shouldn't do two things.

The reason I made this change in this PR rather than another is because `mthdsig` plumbing is dependent on the chain core API's `Head` method, which was removed as part of the goal of this issue in favor of the chain plumbing's `ChainHead` method, so leaving method signatures in the `MessageQuery` return would mean either making `msg` plumbing dependent on `chain` plumbing or moving `MessageQuery` into porcelain, both of which seem significantly more wrong than just dropping this mostly unused return value.

Reply:
> mthdsig should just be part of actor plumbing instead and MessageQuery shouldn't do two things.

hear! hear! fully agree that messagequery was doing too much and should not return the signature. if someone needs it, they can make a call to get it. 

i have more thoughts re plumbing/porcelain organization that will follow with, maybe here or maybe in an issue. i think some of us have different assumptions about what should depend on what and my hypothesis -- sit down for it -- is that this is because we never really talked about it or wrote it down. so lets do that.